src/common/models/measure/measure.ts

```diff
@@ -190,7 +195,7 @@ export class Measure extends BaseImmutable<MeasureValue, MeasureJS> {
     // Expression.some is bugged
     let isApproximate = false;
     this.expression.forEach((exp: Expression) => {
-      if (exp instanceof CountDistinctExpression || exp instanceof QuantileExpression) {
+      if (Measure.isCountDistinctReferences(exp) || Measure.isQuantileReferences(exp)) {
```

Comment:
I think we could remove `this.expression.forEach` and use those two methods. In your code you're doing DFS inside DFS!